# Question - 1

Q1. The weight (in kgs) of a new product launched by Company X is as follows:
[11, 12, 9, 8, 8, 11, 10, 10, 13, 10, 12, 14, 11, 12, 8, 9, 10, 9, 8, 10]
Calculate the confidence interval of the product weight.


In [2]:
import pandas as pd
import numpy as np
import scipy.stats as st
from scipy.stats import sem
import matplotlib.pyplot as plt

In [1]:
# create sample data
data = [11, 12, 9, 8, 8, 11, 10, 10, 13, 10, 12, 14, 11, 12, 8, 9, 10, 9, 8, 10]

print(data)

[11, 12, 9, 8, 8, 11, 10, 10, 13, 10, 12, 14, 11, 12, 8, 9, 10, 9, 8, 10]


In [4]:
# Code here

# 1. Set sample size and confidence level
n = len(data)
confidence = 0.95

# 2. Calculate mean and standard error of the mean (SEM)
mean = np.mean(data)
std_err = st.sem(data)  # Computes s / sqrt(n) automatically

# 4. Calculate the t-distribution confidence interval
lower_bound, upper_bound = st.t.interval(confidence, df=n-1, loc=mean, scale=std_err)

# 5. Output the results
print(f"Sample Mean: {mean:.2f} kg")
print(f"Standard Error (SEM): {std_err:.3f} kg")
print(f"95% Confidence Interval: [{lower_bound:.2f} kg, {upper_bound:.2f} kg]")

Sample Mean: 10.25 kg
Standard Error (SEM): 0.390 kg
95% Confidence Interval: [9.43 kg, 11.07 kg]


To calculate the 95% confidence interval for the product weight, we use a Student's t-distribution because the population standard deviation is unknown and the sample size is small (n = 20).



Step-by-Step Calculation

1. Find the Sample Size (n)
Count of data points: n = 20

2. Calculate the Sample Mean (x-bar)
x-bar = sum(x) / n = (11 + 12 + ... + 10) / 20 = 205 / 20 = 10.25 kg

3. Calculate the Sample Standard Deviation (s)
s = square_root( sum((x - x-bar)^2) / (n - 1) ) approx 1.743 kg

4. Find the Standard Error of the Mean (SEM)
SEM = s / square_root(n) = 1.743 / square_root(20) approx 0.390 kg

5. Find the Critical t-value (t*)
Degrees of freedom (df) = n - 1 = 19
For a 95% confidence level (alpha = 0.05), looking at a t-distribution table: t* = 2.093

6. Calculate the Margin of Error (ME)
ME = t* * SEM = 2.093 * 0.390 approx 0.816 kg

7. Determine the Confidence Interval
CI = x-bar +/- ME = 10.25 +/- 0.816
Lower Limit = 10.25 - 0.816 = 9.434 kg
Upper Limit = 10.25 + 0.816 = 11.066 kg


# Question - 2

Q2.Hospital ABC noted the average weight of the 12-year-old children who come to visit their pediatric ward for a specific type of stomach disease.
weight = [28, 26, 27, 28, 29, 26, 30, 26, 30, 26, 25, 25, 28, 27, 27, 29, 28, 30, 28, 29]
Check the hypothesis that the average weight of kids having this disease is 28 kgs.

In [12]:
# Code here
import numpy as np
import scipy.stats as stats

# 1. Input the dataset
weights = [28, 26, 27, 28, 29, 26, 30, 26, 30, 26, 25, 25, 28, 27, 27, 29, 28, 30, 28, 29]

# 2. Define the hypothesized population mean
hypothesized_mean = 28

# 3. Perform the one-sample t-test
t_statistic, p_value = stats.ttest_1samp(weights, hypothesized_mean)

# 4. Calculate descriptive statistics for context
sample_mean = np.mean(weights)
sample_std = np.std(weights, ddof=1)  # Sample standard deviation (n-1)

# 5. Output results
print(f"Sample Mean: {sample_mean:.2f} kg")
print(f"Sample Standard Deviation: {sample_std:.3f} kg")
print(f"t-statistic: {t_statistic:.3f}")
print(f"p-value: {p_value:.4f}")

# 6. Interpret the results
alpha = 0.05
if p_value < alpha:
    print("Result: Reject the null hypothesis (Significant evidence that average weight is NOT 28 kg).")
else:
    print("Result: Fail to reject the null hypothesis (No significant evidence that average weight differs from 28 kg).")







Sample Mean: 27.60 kg
Sample Standard Deviation: 1.603 kg
t-statistic: -1.116
p-value: 0.2783
Result: Fail to reject the null hypothesis (No significant evidence that average weight differs from 28 kg).


# Question - 3

Q3. Below is the product length (in cms) as processed by two machines.

**sample_1** = [1.873, 1.817, 1.921, 1.945, 1.955, 1.436, 1.960, 1.478, 1.637, 1.869]

**sample_2** = [1.342, 1.432, 1.238, 1.129, 1.146, 1.557, 1.090, 1.483,  1.575, 1.469]


Check whether the two machines produce products with the same or different lengths.  
Assume the following
Observations in each sample are independent and identically distributed (iid).
Observations in each sample are normally distributed.
Observations in each sample have the same variance.

In [ ]:
# Example of the Student's t-test
from scipy.stats import ttest_ind

# ttest_ind: used to calculate the T-test for the means of two independent samples of scores.

# sample data from two different samples
sample_1 = [1.873, 1.817, 1.921, 1.945, 1.955, 1.436, 1.960, 1.478, 1.637, 1.869]
sample_2 = [1.342, 1.432, 1.238, 1.129, 1.146, 1.557, 1.090, 1.483,  1.575, 1.469]

x = ['smaple_1', 'sample_2']
y = [np.mean(sample_1), np.mean(sample_2)]

# Perform independent t-test
t_stat, p_val = stats.ttest_ind(sample_1, sample_2, equal_var=True)
print("T-statistic:", t_stat)
print("P-value:", p_val)

stat=5.174, p=0.000
Probably different distributions


In [9]:

# Example of the Student's t-test
import numpy as np
from scipy.stats import ttest_ind
import scipy.stats as stats

# ttest_ind: used to calculate the T-test for the means of two independent samples of scores.

# sample data from two different samples
sample_1 = [1.873, 1.817, 1.921, 1.945, 1.955, 1.436, 1.960, 1.478, 1.637, 1.869]
sample_2 = [1.342, 1.432, 1.238, 1.129, 1.146, 1.557, 1.090, 1.483,  1.575, 1.469]

x = ['smaple_1', 'sample_2']
y = [np.mean(sample_1), np.mean(sample_2)]

# Perform independent t-test
t_stat, p_val = stats.ttest_ind(sample_1, sample_2, equal_var=True)
print("T-statistic:", t_stat)
print(f"P-value: {p_val:.4f}")
# Interpret the results
alpha = 0.05
if p_val < alpha:
    print("Probably different distributions")
else:
    print("Same Distributions")

T-statistic: 5.173941620022832
P-value: 0.0001
Probably different distributions
